CDTDatastore/HTTP/CDTURLSession.m

```diff
@@ -61,7 +62,12 @@ - (instancetype)initWithCallbackThread:(NSThread *)thread
         if ([[NSURLSessionConfiguration class] respondsToSelector:@selector(backgroundSessionConfigurationWithIdentifier:)]) {
             config = [NSURLSessionConfiguration backgroundSessionConfigurationWithIdentifier:sessionId];
         } else {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+            // since the method is only called on platforms where it's replacement is missing
+            // we supress the warning
```

Comment:
Updated doc in 5955932, it would appear there isn't a way to determine iOS version to use at build time, we have to bite the bullet, or raise our target OS.